### ISSUE TITLE
Data sequence 0xDD, 0xD0

### ISSUE THREAD

**Issue by thomzwg** [toptensoftware/yazd#13]
Not sure if it's my own fault, but lately I've been trying to disassemble a 6 kB binary for MSX. I didn't know in advance which part contained data, so I couldn't use the --entry: option. Yazd only produced a list of Z80 instructions, as expected.

However, one of the data parts happens to have the sequence 0xDD, 0xD0 which does not translate into any Z80 instruction. Yazd disregards the first byte and only gives DEC C.

It's possible to isolate the problem:

```
ORG 0000h

DEFB 0DDh, 0D0h
```

Compiling this and then disassembling using yazd gives:

```
ORG 0000h

DEC C
```

Using a hex editor I know the binary is correct.

**Comment by toptensoftware** [comment]
Thanks for reporting this.  I've just tried this and I get different result to you getting a `RET NC` instruction.

~~~
>yazd test.bin --list
0000: DD D0                             RET     NC
~~~

The disassembler was actually ported [from here](http://z80ex.sourceforge.net/) but I've just gone through it and found this:

https://github.com/toptensoftware/yazd/blob/master/yazd/Disassembler.cs#L140

which looks like if an instruction prefix like (ie: 0xDD) doesn't map to an actual instruction it "mirrors" the instruction as if the prefix wasn't there.

TL;DR: I think it's working correctly.

**Comment by thomzwg** [comment]
I made a typo, 0xD0 should have been 0x0D, that's why you got `RET NC` wheras I got `DEC C`. Thanks for your answer. Although it's working correctly, one byte will disappear because of this.

**Comment by toptensoftware** [comment]
OK.  I still think it's working correctly.  The byte doesn't really disappear, it just becomes a redundant/ignored prefix byte on that instruction:

~~~
yazd test.bin --list
0000: DD 0D                             DEC     C
~~~
